redhat-developer/odo-fork
# Implement a kclient package for kdo

Signed-off-by: John Collier <John.J.Collier@ibm.com>

## What is the purpose of this change? What does it change?
* Implements a `kclient` package for kdo, containing a number of useful kubernetes functions to be used throughout kdo. This will be required for us to start implementing a number of commands in kdo
    * We should be adding additional functions to this package as needed
* Brings over more unit tests from odo: 
    * kclient_test
    * validation_test
    * validators_test
    * preference_test
    * configs_test
* Removes dependencies on `odo` and OpenShift specific packages
    * github.com/openshift/api
    * github.com/openshift/client-go
    * github.com/openshift/odo
    * github.com/redhat-developer/odo

## How to test changes?
* Verify `kdo` builds with `make` or `make install`
* Run the current and new tests with `make test`
